### ISSUE TITLE
Actions failing on Azure webapp

### ISSUE THREAD

**Issue by wes-kay** [actions/checkout#1298]
- Publishing model: Code
- Runtime Stack: Go - 1.19
- Webapp: Basic B2
- Operating System: Linux

Failing on `Deploy to Azure Web App` step and cannot figure out why:

```yaml
name: Go CI/CD Pipeline

on:
  push:
    branches:
      - production

jobs:
  build:
    name: Build
    runs-on: ubuntu-latest
    steps:
    - name: Set up Go
      uses: actions/setup-go@v3
      with:
           go-version: 1.19

    - name: Check out code
      uses: actions/checkout@v2

    - name: Get dependencies
      run: |
        go get -v -t -d ./...    

    - name: Build
      run: |
        go build -o app ./main/main.go

    #failing on this step below
    - name: Deploy to Azure Web App
      uses: azure/webapps-deploy@v2
      with:
        app-name: webapp
        publish-profile: ${{ secrets.AZURE_WEBAPP_PUBLISH_PROFILE }}
```

I am constantly getting different errors but this was the latest:

```
Could not find Gemfile in repo
Could not find typical Ruby files in repo
App in repo is not a Ruby app
Could not find go.mod in repo
Could not find any files with the following extensions in the repo: java, jsp
Platform 'GOLANG' was not detected in the given repository.
Invalid usageMicrosoft.Oryx.BuildScriptGenerator.Exceptions.UnsupportedVersionException: Couldn't detect a version for the platform 'GOLANG' in the repo.
No checker messages emitted
Build Failed
...
Error: Failed to deploy web package to App Service.
Error: Deployment Failed, Package deployment using ZIP Deploy failed. Refer logs for more details.
```
and the web.config:

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <add name="httpPlatformHandler" path="*" verb="*" modules="httpPlatformHandler" resourceType="Unspecified" />
    </handlers>
    <httpPlatform processPath=".\app"
                  arguments=""
                  stdoutLogEnabled="true"
                  stdoutLogFile=".\logs\stdout.log"
                  startupTimeLimit="60">
      <environmentVariables>
        <environmentVariable name="PORT" value="%HTTP_PLATFORM_PORT%" />
      </environmentVariables>
    </httpPlatform>
  </system.webServer>
</configuration>
```
